android/titanium/src/java/org/appcelerator/titanium/TiBaseActivity.java

```diff
@@ -1093,7 +1093,15 @@ private void dispatchCallback(String name, KrollDict data) {
 
 		data.put("source", activityProxy);
 
-		activityProxy.callPropertyAsync(name, new Object[] { data });
+		// TIMOB-19903
+		if (TiApplication.getInstance().runOnMainThread()) {
+			// We must call this synchornously to ensure it happens before we release the Activity reference on the V8/Native side!
+			activityProxy.callPropertySync(name, new Object[] { data });
```

Comment:
That's why I limited the change to only the cases where we're running Kroll on the UI thread - because I wasn't sure if there were other instances of us assuming ordering of these callbacks versus other work. When we're running in that mode, I would assume we'd want to ensure that the onStart would be guaranteed to run while we're doing our onStart() stuff, and not _after_. If we don't run it sync and we're on the UI thread, it's guaranteed as-is to run after we're fully done with all of our setup for the on* calls. When run sync, it's guaranteed to run exactly when we decide to fire this callback in that process.